lib/spack/spack/binary_distribution.py

```diff
@@ -818,16 +819,18 @@ def get_keys(install=False, trust=False, force=False):
             tty.msg("Finding public keys in %s" % mirror_dir)
             files = os.listdir(mirror_dir)
             for file in files:
-                if re.search(r'\.key', file):
+                if re.search(r'\.key', file) or re.search(r'\.pub', file):
                     link = url_util.join(fetch_url_build_cache, file)
                     keys.add(link)
         else:
             tty.msg("Finding public keys at %s" %
                     url_util.format(fetch_url_build_cache))
-            p, links = web_util.spider(fetch_url_build_cache, depth=1)
```

Comment:
@opadron This was different from the way specs are listed so I changed it to use index.html directly. Keys on the e4s mirror are now found where they were not before,